### PR TITLE
Allow For Previous Occurrences To Be Imported When Initiating Class

### PR DESCRIPTION
**Pull request by sampoder**
This would be helpful if you've got a bunch of slugs in a database and you want to make more

### CODE REVIEW

**Comment by sampoder** [comment]
Hopefully that works! :D

File: index.js [hunk 1]
```diff
@@ -9,11 +9,19 @@ export default class BananaSlug {
...
     // eslint-disable-next-line no-unused-expressions
-    this.occurrences
-
+    if(previousOccurrences){
+      this.occurrences = previousOccurrences.reduce((obj, slug) => {
+        const strippedSlug = slug.replace(/-\d+$/, '');
+        obj[strippedSlug] = (obj[strippedSlug] || 0) + 1;
+        return obj;
+      }, {});
+    }
```
Comment (wooorm) [inline index.js:21]: Hey!

- how do you want to save this stuff to a database?
- the `strippedSlug` part comes with a bug: what if someone writes a heading `## Heading 1`?
- I think the `this.reset` below overwrites all your changes
- docs and tests!
- lastly, the previous points can be addressed first, but there’s also a security problem here: a slug of `constructor` will be injected into the object, and cause problems!

---

I’d first like to hear more about your use case.
Perhaps we can come up with a different API. Maybe like a `slugger.toJSON()` function which yields the map, and a `Slugger.fromJSON(map)` to revive it?

**Comment by sampoder** [comment]
Hey @wooorm! Thanks for the review, apologies for the delayed response!

> how do you want to save this stuff to a database?

At the moment, we've got a bunch of rows each with a slug in the slug column.

the strippedSlug part comes with a bug: what if someone writes a heading ## Heading 1?
> I think the this.reset below overwrites all your changes

LOL yes! good catch.

> docs and tests!

certainly, I'll add these once everything else is ready to go

lastly, the previous points can be addressed first, but there’s also a security problem here: a slug of constructor will be injected into the object, and cause problems!

oh fun, what do you exactly mean? a slug that's like "constructor" or something else.

> I’d first like to hear more about your use case.

Basically, we have a bunch of events which each need a unique slug for their page on our site but we want to prevent duplicate slugs.

> Perhaps we can come up with a different API. Maybe like a slugger.toJSON() function which yields the map, and a Slugger.fromJSON(map) to revive it? 

I like this idea! Let me work on creating a new branch and doing this instead.

**Comment by wooorm** [comment]
Do you still need this? Still plan to work on it? How can I help?

**Comment by sampoder** [comment]
Hey! Sorry for the long delay, one day I might get back to this. We no longer need it for this specific project but I do think it'll be cool... hopefully one day I'll be back with a new PR to solve this :D
